src/main.rs

```diff
@@ -3233,6 +3252,70 @@ impl RicCall {
 
             json["NetAccessPoints"] = (*net_access_points).clone();
+            Ok((jsonobj_to_strret(json, req_id), StatusCode::OK))
+        },
+        RicCall::LinkNic => {
+            check_aksk_auth!(auth);
+            let in_json = require_in_json!(bytes);
+            println!("{:#}", in_json.dump());
+
+            let device_number = require_arg!(in_json, "DeviceNumber").as_i32().unwrap();
+            let nic_id = require_arg!(in_json, "NicId");
+            let vm_id = require_arg!(in_json, "VmId");
+            let (vm_idx, vm) = match get_by_id!("Vms", "VmId", vm_id) {
+                Ok((_, idx)) => (idx, &main_json[user_id]["Vms"][idx]),
+                _ => return bad_argument(req_id, json, "Vm not found")
+            };
+            if vm["State"] != "running" && vm["State"] != "stopped" {
+                return bad_argument(req_id, json, "the VM should be either `running` or `stopped`")
+            }
+            if vm.has_key("Nics") {
+                return bad_argument(req_id, json, "the VM is already linked to a Nic")
+            }
+            if !(1..=7).contains(&device_number) {
+                return bad_argument(req_id, json, "the device number should be between 1 and 7, both included")
+            }
+           
+            let nic = match get_by_id!("Nics", "NicId", nic_id) {
+                Ok((_, idx)) => &mut main_json[user_id]["Nics"][idx],
+                _ => return bad_argument(req_id, json, "Nic not found")
+            };
+            link_nic!(nic, device_number);
+            
+            json["LinkNicId"] = nic["LinkNic"]["LinkNicId"].clone();
+            main_json[user_id]["Vms"][vm_idx]["Nics"] = json::array!{ nic.clone() };
+            Ok((jsonobj_to_strret(json, req_id), StatusCode::OK))
+        },
+        RicCall::UnlinkNic => {
+            check_aksk_auth!(auth);
+            let in_json = require_in_json!(bytes);
+            println!("{:#}", in_json.dump());
+
+            let linknic_id = require_arg!(in_json, "LinkNicId");
+            let nic_idx = match main_json[user_id]["Nics"].members_mut().position(|nic| 
+                nic["LinkNic"]["LinkNicId"] == linknic_id && nic["LinkNic"]["DeviceNumber"] != 0) 
+            {
+                Some(idx) => idx,
+                None => return bad_argument(req_id, json, "Nic not found or is a primary Nic")
+            };
+            main_json[user_id]["Nics"][nic_idx].remove("LinkNic");
+
+            let get_vm_niclink_idx = || -> Option<(usize, usize)> {
+                for (vm_idx, vm) in main_json[user_id]["Vms"].members().enumerate() {
+                    if let Some(nic_idx) = vm["Nics"].members().position(|nic| nic["LinkNic"]["LinkNicId"] == linknic_id) {
+                        return Some((vm_idx, nic_idx));
+                    }
+                }
+                None
+            };
+            let (vm_idx, nic_idx) = get_vm_niclink_idx().unwrap();
```

Comment:
add a comment to say that we can't use array_remove, because we need to check Nics len.